collection-scripts/gather_ceph_resources

```diff
@@ -269,7 +261,64 @@ for ns in $namespaces; do
+                images=$(timeout 60 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd ls -p $bp --namespace $rns")
+                pids_rbd=()
+                for image in $images; do
+                    dbglogf "${CEPH_GATHER_DBGLOG}" "collecting vol and snapshot info for ${image}"
+                    {
+                        printf "Collecting image info for: %s/%s%s\n" "${bp}" "${image}" "${rns}"
+                        timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd info $image --pool $bp --namespace $rns" 2>>"${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-"${image}"-debug.log
+                        printf "Collecting image status for: %s/%s%s\n" "${bp}" "${image}" "${rns}"
+                        timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd status $image --pool $bp --namespace $rns" 2>>"${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-status-"${image}"-debug.log
+                        printf "Collecting snap info for: %s/%s%s\n" "${bp}" "${image}" "${rns}"
+                        timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd snap ls --all --format=json --pretty-format $image --pool $bp --namespace $rns" 2>>"${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-snap-json-"${image}"-debug.log
+                    } >>"${COMMAND_OUTPUT_DIR}"/rbd_vol_and_snap_info_"${image}".part &
+                    pids_rbd+=($!)
+                done
+                if [ -n "${pids_rbd[*]}" ]; then
+                    # wait for all pids
+                    dbglog "waiting for ${pids_rbd[*]} to terminate"
+                    wait "${pids_rbd[@]}"
+                fi
+                find "${COMMAND_OUTPUT_DIR}" -name "rbd_vol_and_snap_info_*.part" -print0 | xargs -0 cat >>"${COMMAND_OUTPUT_FILE}"
+                find "${COMMAND_OUTPUT_DIR}" -name "rbd_vol_and_snap_info_*.part" -print0 | xargs -0 rm -f
+                
+                dbglogf "${CEPH_GATHER_DBGLOG}" "Collecting rbd trash ls for rados namespace: %s %s\n" "${bp}" "${rns}"
+                COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/rbd_trash_ls_"${bp}"_"${rns}"
+                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd trash ls --pool $bp --namespace $rns --format=json" >>"${COMMAND_OUTPUT_FILE}"; } >>"${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-trash-ls-"${bp}"-"${rns}"-json-debug.log 2>&1 &
+                
+                COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/rados_rbd_objects_"${bp}"_"${rns}"
+                # List omapkeys in csi.volumes.default in each block pool
+                pvcobjs=$(timeout 60 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rados listomapkeys csi.volumes.default --pool=${bp} --namespace ${rns}")
+                # Get the omap details of each PVC object
+                for pvcobj in $pvcobjs; do
+                    { printf "Name of the pvc object: %s\n" "${pvcobj}" >>"${COMMAND_OUTPUT_FILE}"; }
```

Comment:
@yati1998 is it what we have here is 100% duplication with additional `--namespace` flag. 
IMO if we add any command specific to rbd we need to add it in two different place which is maintenance burden and error prone.

Reply:
As I mentioned in [my previous comment](https://github.com/red-hat-storage/odf-must-gather/pull/150#issuecomment-2077081376), I don't think it's worth jumping through the hoops of having a subfunction, at least for this PR. I'm fine asking to clean that up in a follow-up PR, but as it stands this works and I'd like to move it along.

Reply:
yeah true that, Even I am currently working on it. 
Even if we add functions, we will have to repeat the loops and commands
The only cleanup possible is, adding commands to variables and using those variables at both the places.
Seems just an extra piece of work to me.

@jarrpa I am alreadyon that cleanup, will update the PR, if the collection works fine. orelse @Madhu-1 we can go ahead with this.